examples/render/quad_render/main.rs

```diff
@@ -68,27 +76,10 @@ fn main() {
     let vs_module = device.create_shader_module(include_bytes!("data/vs_main.spv")).unwrap();
     let fs_module = device.create_shader_module(include_bytes!("data/ps_main.spv")).unwrap();
 
-    let set0_layout = device.create_descriptor_set_layout(&[
-            pso::DescriptorSetLayoutBinding {
-                binding: 0,
-                ty: pso::DescriptorType::SampledImage,
-                count: 1,
-                stage_flags: pso::STAGE_FRAGMENT,
-            }
-        ],
-    );
-
-    let set1_layout = device.create_descriptor_set_layout(&[
-            pso::DescriptorSetLayoutBinding {
-                binding: 0,
-                ty: pso::DescriptorType::Sampler,
-                count: 1,
-                stage_flags: pso::STAGE_FRAGMENT,
-            }
-        ],
+    gfx_allocate_descriptor_structs!(context.mut_device();
```

Comment:
hmm, not sure why this has to be a macro?

Reply:
Since you allocate a (potential) batch of descriptor structs, the result can consist of different types and cannot be a vector (unless dynamically typed). We could probably have some tricky tuple imbrication using a builder but it seems worse to use.

Reply:
We could also come up with another descriptor allocation API, obviously. It seemed like an easy way to avoid explicit pools while still allowing some batching.